blockchain/src/blockchain/history_sync.rs

```diff
@@ -2,244 +2,86 @@ use std::error::Error;
-            return Err(PushError::InvalidBlock(BlockError::InvalidHistoryRoot));
-        }
-
-        // Checks if the body exists.
-        let body = macro_block.body.as_ref().ok_or_else(|| {
-            warn!(
-                block = %macro_block,
-                reason = "body missing",
-                "Rejecting block"
-            );
-            PushError::InvalidBlock(BlockError::MissingBody)
-        })?;
-
-        // Check the body root.
-        let body_hash = body.hash::<Blake2bHash>();
-        if macro_block.header.body_root != body_hash {
-            warn!(
-                block = %macro_block,
-                reason = "header body hash doesn't match real body hash",
-                given_body_hash = %macro_block.header.body_root,
-                actual_body_hash = %body_hash,
-                "Rejecting block",
+            log::warn!(
+                "History Chunk failed to verify (chunk {} of epoch {})",
+                chunk_index,
+                self.block.epoch_number()
             );
-            return Err(PushError::InvalidBlock(BlockError::BodyHashMismatch));
+            return Err(PushError::InvalidHistoryChunk);
         }
 
-        // Check the justification.
-        if !TendermintProof::verify(macro_block, &this.current_validators().unwrap()) {
-            warn!(
-                block = %macro_block,
-                reason = "bad justification",
-                "Rejecting block",
-            );
-            return Err(PushError::InvalidBlock(BlockError::InvalidJustification));
-        }
-
-        drop(read_txn);
-
-        // Extend the chain with this block.
-        let prev_macro_info = this.state.macro_info.clone();
-        Blockchain::extend_history_sync(this, block, history, prev_macro_info)
-    }
-
-    /// Extends the current chain with a macro block (election or checkpoint) during history sync.
-    fn extend_history_sync(
-        this: RwLockUpgradableReadGuard<Blockchain>,
-        block: Block,
-        history: &[ExtendedTransaction],
-        mut prev_macro_info: ChainInfo,
-    ) -> Result<PushResult, PushError> {
         // Create a new database write transaction.
-        let mut txn = this.write_transaction();
+        let mut txn = blockchain.write_transaction();
 
         // Get the block hash.
-        let block_hash = block.hash();
+        let current_batch = policy::batch_at(self.block.block_number());
 
         // Calculate the cumulative transaction fees for the given batch. This is necessary to
         // create the chain info for the block.
-        let mut cum_tx_fees = Coin::ZERO;
-        let current_batch = policy::batch_at(block.block_number());
-        for i in (0..history.len()).rev() {
-            if policy::batch_at(history[i].block_number) != current_batch {
+        for i in (0..history_chunk.history.len()).rev() {
+            if policy::batch_at(history_chunk.history[i].block_number) != current_batch {
                 break;
```

Comment:
Added a comment to clarify it but essentially they are ignored since we only need them for the same batch the macro block is at to later update the `ChainInfo` in the `Blockchain`